corehq/apps/app_manager/models.py

```diff
@@ -3508,6 +3499,21 @@ def get_filter_value(self, user, ui_filter):
         return None
 
 
+class AncestorLocationTypeFilter(ReportAppFilter):
+    ancestor_location_type_name = StringProperty()
+
+    def get_filter_value(self, user, ui_filter):
+        from corehq.apps.locations.models import SQLLocation
+
+        try:
+            ancestor = user.sql_location.get_ancestors(include_self=True).\
+                get(location_type__name=self.ancestor_location_type_name)
+        except AttributeError, SQLLocation.DoesNotExist:
```

Comment:
this needs parens around the exceptions to do what you want (i'm actually a little surprised that this works at all given that the second statement should be an assignment.)